src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -82,7 +82,20 @@ public JsonSerializerOptions(JsonSerializerOptions options)
             // _classes is not copied as sharing the JsonClassInfo and JsonPropertyInfo caches can result in
             // unnecessary references to type metadata, potentially hindering garbage collection on the source options.
 
-            // _haveTypesBeenCreated is not copied; it's okay to make changes to this options instance as (de)serialization has not occured.
+            // _haveTypesBeenCreated is not copied; it's okay to make changes to this options instance as (de)serialization has not occurred.
+        }
+
+        /// <summary>
+        /// Constructs a new <see cref="JsonSerializerOptions"/> instance with predefined set of options determined by <see cref="JsonSerializerDefaults"/>.
+        /// </summary>
+        /// <param name="defaults"> The <see cref="JsonSerializerDefaults"/> to reason about.</param>
+        public JsonSerializerOptions(JsonSerializerDefaults defaults) : this()
+        {
+            if (defaults == JsonSerializerDefaults.Web)
+            {
+                _propertyNameCaseInsensitive = true;
+                _jsonPropertyNamingPolicy = JsonNamingPolicy.CamelCase;
```

Comment:
@layomia - asked me to drop by and confirm whether this matches ASP.NET's expectations.  The relevant code is [here](https://github.com/dotnet/aspnetcore/blob/master/src/Mvc/Mvc.Core/src/JsonOptions.cs#L29) for anyone who wants to compare.

Some items worth of discussion:

- `MaxDepth`
- *relaxed encoder*

For MaxDepth - we did not scientifically compute the number for maxdepth. We have just always had a value of 32 - and yours is 64. I don't foresee a problem with us using 64. So I think we can agree to use your default of 64.

For the *encoder* - (this is the fun one) we do some pretty subtle stuff. Our main options class doesn't specify an encoder, and we special case it in two places:

- For writing output as JSON response from the server [we opt in the *relaxed* encoder](https://github.com/dotnet/aspnetcore/blob/4191eb5c7629ddbf88168ac45cffb7bea3b7e956/src/Mvc/Mvc.Core/src/Formatters/SystemTextJsonOutputFormatter.cs)
- For writing [JSON in HTML](https://github.com/dotnet/aspnetcore/blob/4191eb5c7629ddbf88168ac45cffb7bea3b7e956/src/Mvc/Mvc.ViewFeatures/src/Rendering/SystemTextJsonHelper.cs#L29) we force the default (strict) encoder 

If we want to be technical - JSON as a document/request/response with the content type `application/json` is always unicode. The JSON specs all agree, that JSON is all unicode all of the time. We use the relaxed encoder when sending JSON over the wire as its own self-contained payload because all consumers should expect it to the unicode.

We *don't* use the relaxed encoder when JSON is being embedded inside an HTML response, because that's browser-land, and it's much more of a wierd place.

I think regardless of which encoder is used as the default we'll keep our behavior the same. I won't try to cosplay as a security expert. My assumption is that the common task using the "web" settings is to read and write JSON documents for communication with a REST API, so my intuition is that using the relaxed encoding is right.

/cc @GrabYourPitchforks @pranavkm

Reply:
> My assumption is that the common task using the "web" settings is to read and write JSON documents for communication with a REST API, so my intuition is that using the relaxed encoding is right.

I agree with you on that. Although, just `Web` option would mislead code clients to potential vulnerability issues. We could provide a more verbose name that communicates usage and intentions. Something like `DO_NO_USE_IT_IN_HTML__Web` JK

Reply:
I discussed with @terrajobst - since specifying the unsafe relaxed encoder isn't recommended in some scenarios, we shouldn't specify it in a generic "web" default. Developers should specify it atop the default, depending on the scenario, as is done in an example @rynowak provided above. So, the current implementation works fine.

We don't want to have various "web" default variants (WebRest, WebHtml etc) as this would likely introduce the overhead of figuring out what the differences are, particularly if the set of defaults grows.

We'll leave max depth (effective) as 64, since we don't foresee issues with ASP.NET switching to this number.